app/services/fastpass/FastPassService.scala

```diff
@@ -82,6 +87,19 @@ trait FastPassService extends EventSink {
     } yield ()
   }
 
+  private def autoProgressToSiftOrFSAC(applicationId: String): Future[Unit] = {
+    for {
+      preferences <- schemePreferencesService.find(applicationId)
+    } yield {
+      val hasSiftableScheme = schemesRepository.siftableSchemeIds.intersect(preferences.schemes).isEmpty
+      if(hasSiftableScheme) {
+        appRepo.addProgressStatusAndUpdateAppStatus(applicationId, ProgressStatuses.SIFT_ENTERED)
+      } else {
+        appRepo.addProgressStatusAndUpdateAppStatus(applicationId, ProgressStatuses.ASSESSMENT_CENTRE_AWAITING_ALLOCATION)
```

Comment:
I suspect this is returning a Future[Future[Unit]] - which means you'll have an unmapped future without a .flatMap(identity)

Reply:
The way Future[Unit] eats other futures makes me so sad :-(

Reply:
This is very strange behaviour. I'm very curious why this happens(perhaps some weird type erasure). Thanks for the catch